category/views.py

```diff
@@ -1,7 +1,21 @@
 from base.mixins import CompanyFilterMixin
-from base.views import RetrieveCreateUpdateDestroyView
+from base.views import ListView, RetrieveCreateUpdateDestroyView
 
+from .serializers import CategorySerializer
+from .models import Category
 
-class CategoryView(CompanyFilterMixin, RetrieveCreateUpdateDestroyView):
+
+class CategoryMixin(CompanyFilterMixin):
+    lookup_field = 'name'
```

Comment:
Looking up by name can be difficult. You have to make sure that the transaction serializer returns the name, not the ID, and you have to think about what happens if a name is changed